include/netlicensing/validation_parameters.h

```diff
@@ -56,12 +71,20 @@ namespace netlicensing {
     */
     [[deprecated("use NodeLocked licensing model instead")]]
     void setLicenseeSecret(const String_t& licenseeSecret) {
-      licenseeSecret_i = licenseeSecret;
+        licenseeParameters_i[PROP_LICENSEE_SECRET] = licenseeSecret.toString();
     }
 
     [[deprecated("use NodeLocked licensing model instead")]]
     const String_t& getLicenseeSecret() const {
-      return licenseeSecret_i;
+        String_t licenseeSecret;
+
+        auto it = licenseeParameters_i.find(PROP_LICENSEE_SECRET);
+
+        if (it != licenseeParameters_i.end()) {
+            licenseeSecret = (String_t)it->second;
+        }
+
+        return licenseeSecret;
```

Comment:
Again, returning local by reference. This may not be clearly visible when using MSVC toolset, as it has some non-standard language extensions. Try to build additionally with GCC toolset (e.g. in a linux docker container).